contrib/babelfishpg_common/Makefile

```diff
@@ -87,8 +88,24 @@ $(EXTENSION).control: $(EXTENSION).control.in
 		| sed -e 's|@MODULEPATH@|$(MODULEPATH)|g' \
 		> $@
 
-sql/$(EXTENSION)--$(EXTVERSION).sql: sql/$(EXTENSION).in
-	cpp $< | sed 's/^# /-- /g' > $@
+FLAG_TO_CHECK := -DENABLE_SPATIAL_TYPES
+
+# Check if the flag is present in PG_CPPFLAGS
+ifeq (,$(filter $(FLAG_TO_CHECK),$(PG_CPPFLAGS)))
+# The flag is not present then build the .in file not including the spatial type files
+    sql/$(EXTENSION)--$(EXTVERSION).sql: sql/$(EXTENSION).in
+		cpp $< | sed 's/^# /-- /g' > $@
+
+    sql/upgrades/$(EXTENSION)--3.2.0--3.3.0.sql: sql/upgrades/babelfishpg_upgrades.in
```

Comment:
Usually, it is not a good idea to include hard-coded version numbers in Makefile because we need to update this accordingly for every new versions. Have you considered a way not including the version number here?

Reply:
For Upgrade file, now onwards we need to add a line for each upgrade file here
eg:  ```sql/upgrades/$(EXTENSION)--3.2.0--3.3.0.sql: sql/upgrades/babelfishpg_upgrades.in```   (for babelfishpg_common--3.2.0--3.3.0.sql
eg:  ```sql/upgrades/$(EXTENSION)--3.3.0--3.4.0.sql: sql/upgrades/babelfishpg_upgrades.in```   (for babelfishpg_common--3.3.0--3.4.0.sql
so that we could build all the upgrade files, and not just a particular version upgrade file.
for eg: if we keep a variable ```sql/upgrades/$(EXTENSION)--PREV_VERSION--CUR_VERSION.sql: sql/upgrades/babelfishpg_upgrades.in``` then it will only create a particular upgrade file corresponding to the values provided, but we need all the upgrade files here so, we will add a line for each new upgrade file to create with hardcoded versions.